targets/arty/net.py

```diff
@@ -31,7 +31,7 @@ def __init__(self, platform, *args, **kwargs):
             platform.request("eth_clocks"),
             platform.request("eth"))
         self.submodules.ethmac = LiteEthMAC(
```

Comment:
I don't quite understand why LiteEthMAC needs to understand the cpu endianness? However it looks like that is something from upstream, not something you invented?

Reply:
The MAC converts 32-bit writes/reads to/from 8-bit streams so needs to know the endiannness to know in which order reading/storing the streams.

Reply:
I have detected the problems when testing the firmware on HW. Without setting the proper endianess, data read from `liteeth` buffers had each 4-byte chunk reverted, causing the software to fail to detect the proper type of packet and drop it in a process.

After realising what was going on, I tested the pure LiteX platform and it worked fine. Diffing them pointed to the `endianness` property (not set in LiteX-BuildEnv).